tests/cpu/test_lazy_reorder.py

```diff
@@ -641,6 +641,52 @@ def test_adaptive_avg_pool2d_backward(self):
         y_cpu.backward()
         y_dpcpp.backward()
         self.assertEqual(x_cpu.grad, x_dpcpp.grad)
+    
+    def test_adaptive_avg_pool2d_not_divisible(self):
+        ipex.enable_auto_dnnl()
+        rand_seed = int(get_rand_seed())
+        print("{} rand sed: {}".format(sys._getframe().f_code.co_name, rand_seed))
+        torch.manual_seed(rand_seed)
+        N = torch.randint(3, 10, (1,)).item()
+        C = torch.randint(3, 10, (1,)).item()
+        x_cpu = torch.randn(N, C, 224, 224, dtype=torch.float32) * 100
+        x_dpcpp = x_cpu.to(device=device)
+        # test the fallback to cpu when the input size is not divisible by the output size
+        adaptive_avg_pool2d = torch.nn.AdaptiveAvgPool2d(6)
+
+        y_cpu = adaptive_avg_pool2d(x_cpu)
+        y_dpcpp = adaptive_avg_pool2d(x_dpcpp)
+
+        self.assertEqual(
+            y_cpu,
+            y_dpcpp)
+
+        self.assertEqual("cpu", y_cpu.device.type)
+        self.assertEqual(device, y_dpcpp.device.type)
+
+    def test_adaptive_avg_pool2d_backward_not_divisible(self):
+        ipex.enable_auto_dnnl()
```

Comment:
is it possible to merge the forward and backward ut into a single one?

Reply:
Other tests in test_lazy_reorder.py separates the backward from the forward. Should we align with other ut in this file? https://github.com/intel/intel-extension-for-pytorch/blob/1d432c2414889e89231c94c7c80d80930c1792b9/tests/cpu/test_lazy_reorder.py#L711

Reply:
okay, that's just fine.
This code was copied from `test_mkldnn.py` where we've written lots of copy-pasted code there.